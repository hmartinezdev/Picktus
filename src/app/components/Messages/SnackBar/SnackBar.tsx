import colors from '@constants/colors';
import { borderRadius, fontFamily } from '@constants/styles';
import { PicktusMessageLevel } from '@store/reducers/message';
import React, { Component } from 'react';
import MessageIcon from '../MessageIcon';

export interface ISnackBarProps {
  text: string;
  level: PicktusMessageLevel;
}

class SnackBar extends Component<ISnackBarProps> {
  public render(): React.ReactElement<SnackBar> {
    const { text, level } = this.props;
    return (
      <div>
        <div className={`snackbar snackbar--${level}`}>
          <MessageIcon level={level} />
          <p className="text">{text}</p>
        </div>
        <style jsx>{`
          .snackbar {
            display: flex;
            justify-content: center;
            align-items: center;
            font-family: ${fontFamily};
            background: ${colors.primary};
            padding: 1rem;
            color: ${colors.white};
            font-size: 1rem;
            box-shadow: 0 0 25px rgba(33, 33, 33, 0.5);
            border-radius: ${borderRadius};
            box-sizing: border-box;
            animation: 0.3s down ease 1;
          }

          .text {
            height: 40px;
            line-height: 40px;
            padding: 0 10px;
            text-align: center;
          }

          .snackbar--${PicktusMessageLevel.ERROR} {
            background-color: ${colors.error};
            color: ${colors.white};
          }

          .snackbar--${PicktusMessageLevel.SUCCESS} {
            background-color: ${colors.green};
            color: ${colors.white};
          }

          .snackbar--${PicktusMessageLevel.WARNING} {
            background-color: ${colors.warning};
            color: ${colors.black};
          }
        `}</style>
      </div>
    );
  }
}

export default SnackBar;
