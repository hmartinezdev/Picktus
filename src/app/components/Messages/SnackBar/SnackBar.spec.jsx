import { shallow } from 'enzyme';
import React from 'react';
import SnackBar from './SnackBar';

let wrapper;

const setup = () => shallow(<SnackBar />);

describe('<SnackBar />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
