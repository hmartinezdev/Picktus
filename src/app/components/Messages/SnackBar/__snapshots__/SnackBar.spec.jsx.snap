// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnackBar /> should render properly 1`] = `
<div>
  <div
    className="snackbar snackbar--undefined"
  >
    <MessageIcon
      height={32}
      width={32}
    />
    <p
      className="text"
    />
  </div>
  <style
    jsx={true}
  >
    
          .snackbar {
            display: flex;
            justify-content: center;
            align-items: center;
            font-family: 'Josefin Sans', sans-serif;
            background: #2d3047;
            padding: 15px;
            color: #F8F8F7;
            font-size: 16px;
            box-shadow: 0 0 25px rgba(33, 33, 33, 0.5);
            border-radius: 3px;
            box-sizing: border-box;
            animation: 0.3s down ease 1;
          }

          .text {
            height: 40px;
            line-height: 40px;
            padding: 0 10px;
            text-align: center;
          }

          .snackbar--ERROR {
            background-color: #982649;
            color: #F8F8F7;
          }

          .snackbar--SUCCESS {
            background-color: #75ab88;
            color: #F8F8F7;
          }

          .snackbar--WARNING {
            background-color: #E37E5B;
            color: #151614;
          }
        
  </style>
</div>
`;
